src/services/completions.ts

```diff
@@ -371,6 +381,8 @@ namespace ts.Completions {
                 }
             }
 
+            result = result.filter(entry => entry.name.indexOf(baseName) >= 0);

```

Comment:
Removing the filtering altogether.

Per our offline conversation, I found that VS and VS Code provide more detailed completions such as camel-case based ones (for VS Code, viewable at [src/vs/base/common/filters.ts](https://github.com/Microsoft/vscode/blob/master/src/vs/base/common/filters.ts)) and so I'm deferring the work of filtering to the client, as appears to be the norm in the status quo.